### ISSUE TITLE
Add function to get current active layout on a page

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#791]
I'm doing a block module (https://github.com/docwilmot/fanciblock), drastic port of https://www.drupal.org/project/iconbox, and I'm trying to put settings for block instances in config. I need to create a programmatic link to a block settings form in layouts, but this is built per layout like `/admin/structure/layouts/manage/<layout>/configure-redirect/<UUID>?destination=node#configure-block:<UUID>`
I cannot figure how to get the Layout either on block view nor on block creation. Is there a way? 
It seems we need a few more layout API functions than we've got.

**Comment by quicksketch** [comment]
I think you're right. I can't find a way to get the current layout either.

**Comment by arshad** [comment]
Not sure if this is going to help but here are some helpful layout functions:

### Load a layout by path
```
layout_load_multiple_by_path($path);
```

### Load a layout by name
```
layout_load($layout_name)
```
Think you can get the name of the layout using arg from the path and load the layout using this?

### Get the default layout
```
layout_load('default');
```

### Get the default admin layout
```
layout_load('admin_default');
```

**Comment by docwilmot** [comment]
New PR Up

**Comment by docwilmot** [comment]
Updated to static cache. Force push new commit.

**Comment by quicksketch** [comment]
The new PR at  adds a function for `layout_get_layout_by_path()`, which acts a lot like `layout_load_multiple_by_path()` except it does the job of actually selecting a layout. Now with the static caching, it can be used to get the current layout without any additional processing.

**Comment by quicksketch** [comment]
Thanks @docwilmot for the PR! This is now merged for 1.4.0.
